src/transport/manager/address.rs

```diff
@@ -259,6 +259,13 @@ impl AddressStore {
         records.sort_by(|lhs, rhs| rhs.score.cmp(&lhs.score));
         records.into_iter().take(limit).map(|record| record.address).collect()
     }
+
+    /// Similar to [`Self::addresses`], but returns an iterator over the addresses.
+    pub fn addresses_iter(&self) -> impl Iterator<Item = &Multiaddr> {
+        let mut records = self.addresses.values().collect::<Vec<_>>();
+        records.sort_by(|lhs, rhs| rhs.score.cmp(&lhs.score));
+        records.into_iter().map(|record| &record.address)
+    }
```

Comment:
Returning an iterator after allocating a vector "hides" the cost of calling this function. IMO having `addresses()` should be enough.